benchmarks/dynamo/common.py

```diff
@@ -236,13 +273,30 @@ def print_summary(filename):
             pass
 
 
+def tensor_is_on_xla(tensors):
+    if not isinstance(tensors, (tuple, list)):
+        tensors = [tensors]
+    return any(map(lambda x: x.device.type == "xla", tensors))
+
+
 def timed(model, model_iter_fn, example_inputs, times=1, return_result=False):
     synchronize()
     reset_rng_state()
     t0 = time.perf_counter()
     # Dont collect outputs to correctly measure timing
     for _ in range(times):
         result = model_iter_fn(model, example_inputs, collect_outputs=False)
+        if tensor_is_on_xla(result):
```

Comment:
the check is loose (any) - should it be tight? (all)

Reply:
yep, but any valid case should not mix xla/eager tensors. So for valid cases, checking 'all' or 'any' should be equivalent. I could add assertion to guard against invalid (mix tensor type) cases. Let me know if you want me to do that.